packages/search-ui/src/SearchDriver.js

```diff
@@ -219,85 +222,147 @@ export default class SearchDriver {
       ...searchParameters
     };
 
+    // We want to make sure if this method is running, that any calls to this
+    // method that had been previously deferred are cancelled. We don't want to
+    // possibly have the deferred call execute after this immediate call.
+    this.debounceManager.cancelByName("_updateSearchResults");
+
+    // We bail on making state updates if "isLoading" is true, which implies
+    // that there is an active API request that has been made and is pending.
+    //
+    // The net effect of this is that the user interaction effectively becomes
+    // "locked" while API requests are in flight, to avoid any odd race
+    // conditions.
+    //
+    // NOTE: This may be unnecessary now that we have debounce
+    // "_makeSearchRequest".
     if (isLoading && !ignoreIsLoadingCheck) return;
 
     this._setState({
       current,
       error: "",
       filters,
-      isLoading: true,
       resultsPerPage,
       searchTerm,
       sortDirection,
       sortField
     });
 
-    const requestId = this.requestSequencer.next();
-
-    const queryConfig = {
-      ...this.searchQuery,
-      facets: removeConditionalFacets(
-        this.searchQuery.facets,
-        this.searchQuery.conditionalFacets,
-        filters
-      )
-    };
-
-    const requestState = filterSearchParameters(this.state);
-
-    return this.events.search(requestState, queryConfig).then(
-      resultState => {
-        if (this.requestSequencer.isOldRequest(requestId)) return;
-        this.requestSequencer.completed(requestId);
+    this._makeSearchRequest({
+      current,
+      filters,
+      resultsPerPage,
+      searchTerm,
+      skipPushToUrl,
+      sortDirection,
+      sortField
```

Comment:
For keys that are already in state due to `this.setState` above - do we want to consider just pulling those from `this.state` instead of passing `current`, `filters`, `resultsPerPage`, `searchTerm`, `sortDirection`, and `sortField` twice?

Reply:
Yes, I think that would be more clear, and also let _makeSearchRequest callable from other areas of code directly.